<a href="https://colab.research.google.com/github/Archana310/econometrics-data/blob/main/Assignment_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

CONSTRUCTING CAPM MODEL OF INDIAN STOCK EXCHANGE DATA OF NIFTY & CIPLA

1.Install yfinance (yahoo finance) in colab notebook

In [32]:
! pip install yfinance

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


import yfinance library for importing NIFTY index data & CIPLA stock data

In [33]:
import yfinance as yf

3.importing NIFTY index data from yfinance by using ticker "^NSEI"

In [34]:
NIFTY=yf.Ticker('^NSEI')

4.import 5 year NIFTY & CIPLA data (only close values), daily basis convert it into monthly % return basis

In [35]:
NIFTY_d=NIFTY.history(period='5y',auto_adjust="True")['Close']

In [36]:
NIFTY_d

Date
2017-05-29     9604.900391
2017-05-30     9624.549805
2017-05-31     9621.250000
2017-06-01     9616.099609
2017-06-02     9653.500000
                  ...     
2022-05-23    16214.700195
2022-05-24    16125.150391
2022-05-25    16025.799805
2022-05-26    16170.150391
2022-05-27    16334.150391
Name: Close, Length: 1232, dtype: float64

In [37]:
NIFTY_m=NIFTY_d.resample('1M').last()

In [38]:
NIFTY_m

Date
2017-05-31     9621.250000
2017-06-30     9520.900391
2017-07-31    10077.099609
2017-08-31     9917.900391
2017-09-30     9788.599609
                  ...     
2022-01-31    17339.849609
2022-02-28    16793.900391
2022-03-31    17464.750000
2022-04-30    17102.550781
2022-05-31    16334.150391
Freq: M, Name: Close, Length: 61, dtype: float64

In [39]:
NIFTY_mr=NIFTY_m.pct_change().dropna()

In [40]:
NIFTY_mr

Date
2017-06-30   -0.010430
2017-07-31    0.058419
2017-08-31   -0.015798
2017-09-30   -0.013037
2017-10-31    0.055851
2017-11-30   -0.010522
2017-12-31    0.029741
2018-01-31    0.047195
2018-02-28   -0.048501
2018-03-31   -0.036134
2018-04-30    0.061862
2018-05-31   -0.000298
2018-06-30   -0.002035
2018-07-31    0.059939
2018-08-31    0.028530
2018-09-30   -0.064214
2018-10-31   -0.049756
2018-11-30    0.047191
2018-12-31   -0.001306
2019-01-31   -0.002909
2019-02-28   -0.003550
2019-03-31    0.072041
2019-04-30    0.015398
2019-05-31    0.014866
2019-06-30   -0.011235
2019-07-31   -0.056905
2019-08-31   -0.008522
2019-09-30    0.040932
2019-10-31    0.035122
2019-11-30    0.015037
2019-12-31    0.009323
2020-01-31   -0.016958
2020-02-29   -0.063563
2020-03-31   -0.232464
2020-04-30    0.146800
2020-05-31   -0.028357
2020-06-30    0.075342
2020-07-31    0.074873
2020-08-31    0.028361
2020-09-30   -0.012290
2020-10-31    0.035105
2020-11-30    0.113941
2020-12-31    0.078094
2021-0

In [41]:
CIPLA=yf.Ticker('CIPLA.NS')

In [42]:
CIPLA_d=CIPLA.history(period='5y',auto_adjust="True")['Close']

In [43]:
CIPLA_d

Date
2017-05-29    494.218323
2017-05-30    499.254883
2017-05-31    504.976013
2017-06-01    504.193573
2017-06-02    518.374207
                 ...    
2022-05-23    970.049988
2022-05-24    970.400024
2022-05-25    971.849976
2022-05-26    964.549988
2022-05-27    965.349976
Name: Close, Length: 1236, dtype: float64

In [44]:
CIPLA_m=CIPLA_d.resample('1M').last()

In [45]:
CIPLA_m

Date
2017-05-31     504.976013
2017-06-30     543.361389
2017-07-31     549.710266
2017-08-31     561.291443
2017-09-30     575.227905
                 ...     
2022-01-31     945.000000
2022-02-28     925.049988
2022-03-31    1018.049988
2022-04-30     981.200012
2022-05-31     965.349976
Freq: M, Name: Close, Length: 61, dtype: float64

In [46]:
CIPLA_mr=CIPLA_m.pct_change().dropna()

In [47]:
CIPLA_mr

Date
2017-06-30    0.076014
2017-07-31    0.011684
2017-08-31    0.021068
2017-09-30    0.024829
2017-10-31    0.070807
2017-11-30   -0.042543
2017-12-31    0.012648
2018-01-31   -0.026869
2018-02-28   -0.004222
2018-03-31   -0.074960
2018-04-30    0.113576
2018-05-31   -0.135989
2018-06-30    0.175019
2018-07-31    0.040055
2018-08-31    0.037443
2018-09-30   -0.012233
2018-10-31   -0.037918
2018-11-30   -0.140485
2018-12-31   -0.039475
2019-01-31   -0.004235
2019-02-28    0.072009
2019-03-31   -0.052205
2019-04-30    0.074962
2019-05-31   -0.010885
2019-06-30   -0.009663
2019-07-31   -0.052988
2019-08-31   -0.093360
2019-09-30   -0.099376
2019-10-31    0.097180
2019-11-30   -0.000321
2019-12-31    0.024641
2020-01-31   -0.065454
2020-02-29   -0.100246
2020-03-31    0.054336
2020-04-30    0.394348
2020-05-31    0.099305
2020-06-30   -0.012188
2020-07-31    0.124795
2020-08-31   -0.009165
2020-09-30    0.085698
2020-10-31   -0.026075
2020-11-30   -0.011796
2020-12-31    0.099718
2021-0

5.As we imported NIFTY & CIPLA monthly % change basis data, Now we have to merge NIFTY & CIPLA data together on the basis of "Date". For the same purpose we need to import pandas

In [48]:
import pandas as pd

In [49]:
CN_dt=pd.merge(NIFTY_mr,CIPLA_mr, on='Date')

In [50]:
CN_dt.head()

Close_x   Close_y
Date                          
2017-06-30 -0.010430  0.076014
2017-07-31  0.058419  0.011684
2017-08-31 -0.015798  0.021068
2017-09-30 -0.013037  0.024829
2017-10-31  0.055851  0.070807

In [51]:
CN_dt.rename(columns={'Close_x':'NIFTY_mr','Close_y':'CIPLA_mr'},inplace=True)

In [52]:
CN_dt

NIFTY_mr  CIPLA_mr
Date                          
2017-06-30 -0.010430  0.076014
2017-07-31  0.058419  0.011684
2017-08-31 -0.015798  0.021068
2017-09-30 -0.013037  0.024829
2017-10-31  0.055851  0.070807
2017-11-30 -0.010522 -0.042543
2017-12-31  0.029741  0.012648
2018-01-31  0.047195 -0.026869
2018-02-28 -0.048501 -0.004222
2018-03-31 -0.036134 -0.074960
2018-04-30  0.061862  0.113576
2018-05-31 -0.000298 -0.135989
2018-06-30 -0.002035  0.175019
2018-07-31  0.059939  0.040055
2018-08-31  0.028530  0.037443
2018-09-30 -0.064214 -0.012233
2018-10-31 -0.049756 -0.037918
2018-11-30  0.047191 -0.140485
2018-12-31 -0.001306 -0.039475
2019-01-31 -0.002909 -0.004235
2019-02-28 -0.003550  0.072009
2019-03-31  0.072041 -0.052205
2019-04-30  0.015398  0.074962
2019-05-31  0.014866 -0.010885
2019-06-30 -0.011235 -0.009663
2019-07-31 -0.056905 -0.052988
2019-08-31 -0.008522 -0.093360
2019-09-30  0.040932 -0.099376
2019-10-31  0.035122  0.097180
2019-11-30  0.015037 -0.000321
2019-12-31  0.009323  0.024641
2020-01-31 -0.016958 -0.065454
2020-02-29 -0.063563 -0.100246
2020-03-31 -0.232464  0.054336
2020-04-30  0.146800  0.394348
2020-05-31 -0.028357  0.099305
2020-06-30  0.075342 -0.012188
2020-07-31  0.074873  0.124795
2020-08-31  0.028361 -0.009165
2020-09-30 -0.012290  0.085698
2020-10-31  0.035105 -0.026075
2020-11-30  0.113941 -0.011796
2020-12-31  0.078094  0.099718
2021-01-31 -0.024829  0.007257
2021-02-28  0.065609 -0.047040
2021-03-31  0.011119  0.035639
2021-04-30 -0.004057  0.116857
2021-05-31  0.065046  0.042841
2021-06-30  0.008901  0.023753
2021-07-31  0.002643 -0.053349
2021-08-31  0.086858  0.035845
2021-09-30  0.028365  0.037719
2021-10-31  0.003037 -0.079813
2021-11-30 -0.038958  0.073200
2021-12-31  0.021836 -0.028004
2022-01-31 -0.000818  0.000953
2022-02-28 -0.031485 -0.021111
2022-03-31  0.039946  0.100535
2022-04-30 -0.020739 -0.036197
2022-05-31 -0.044929 -0.016154

6.Now as our NIFTY_mr & CIPLA_mr last 5 years data is ready for analysis, Now we can set up CAPM model including regression as follows

Y = f (X)

Y = b0 + (b1*X)

Y = CIPLA_mr X = NIFTY_mr

for this purpose we will first import statsmodels

In [53]:
import statsmodels.formula.api as smf

In [54]:
reg='CIPLA_mr~NIFTY_mr'

In [55]:
regoutput=smf.ols(reg,CN_dt).fit()

In [56]:
print(regoutput.summary())

                            OLS Regression Results                            
Dep. Variable:               CIPLA_mr   R-squared:                       0.073
Model:                            OLS   Adj. R-squared:                  0.057
Method:                 Least Squares   F-statistic:                     4.597
Date:                Fri, 27 May 2022   Prob (F-statistic):             0.0362
Time:                        09:29:49   Log-Likelihood:                 66.876
No. Observations:                  60   AIC:                            -129.8
Df Residuals:                      58   BIC:                            -125.6
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0097      0.011      0.915      0.3

Here we get regression output of our CAPM model which says that NIFTY_mr(X) & CIPLA_mr(Y) are having following relationship

Y = b0 + (b1*X)

With the help of regression analysis we got following beta values

b0 = 0.0097 b1 = 0.4139

So, our CAPM model is

Y = 0.0097 + (0.4139*X)